Build fails with upcoming Boost 1.89.0

In the upcoming Boost 1.89.0 release, the Boost.System stub library introduced back in 1.69[^1] has been removed (https://github.com/boostorg/system/commit/7a495bb46d7ccd808e4be2a6589260839b0fd3a3), which causes CMake error:
```
CMake Error at /opt/homebrew/lib/cmake/Boost-1.89.0/BoostConfig.cmake:141 (find_package):
  Could not find a package configuration file provided by "boost_system"
  (requested version 1.89.0) with any of the following names:

    boost_systemConfig.cmake
    boost_system-config.cmake

  Add the installation prefix of "boost_system" to CMAKE_PREFIX_PATH or set
  "boost_system_DIR" to a directory containing one of the above files.  If
  "boost_system" provides a separate development package or SDK, be sure it
  has been installed.
Call Stack (most recent call first):
  /opt/homebrew/lib/cmake/Boost-1.89.0/BoostConfig.cmake:262 (boost_find_component)
  /opt/homebrew/opt/cmake/share/cmake/Modules/FindBoost.cmake:609 (find_package)
  CMakeLists.txt:50 (find_package)
```

According to `https://github.com/boostorg/system/issues/132#issuecomment-3146378680`, common fixes include:
* increasing minimum to 1.69 and drop `system` component, or
* keep current minimum (1.68) and use OPTIONAL_COMPONENTS

[^1]: https://www.boost.org/doc/libs/1_69_0/libs/system/doc/html/system.html#changes_in_boost_1_69

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -41,7 +41,6 @@ jobs:
         libboost-filesystem-dev \
         libboost-program-options-dev \
         libboost-serialization-dev \
-        libboost-system-dev \
         libboost-test-dev \
         libeigen3-dev \
         libflann-dev \
@@ -58,7 +57,7 @@ jobs:
       with:
           vcpkgGitCommitId: da4b78d35084ec4e9808aa02017028da03b8d2ab
           vcpkgTriplet: 'x64-windows'
-          vcpkgArguments: 'boost-dynamic-bitset boost-filesystem boost-graph boost-odeint boost-program-options boost-serialization boost-system boost-test eigen3 flann'
+          vcpkgArguments: 'boost-dynamic-bitset boost-filesystem boost-graph boost-odeint boost-program-options boost-serialization boost-test eigen3 flann'
 
     - name: Build & Test (Linux, macOS)
       if: runner.os != 'Windows'

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -47,7 +47,7 @@ set_package_properties(Boost PROPERTIES
     URL "https://www.boost.org"
     PURPOSE "Used throughout OMPL for data serialization, graphs, etc.")
 set(Boost_USE_MULTITHREADED ON)
-find_package(Boost 1.68 REQUIRED COMPONENTS serialization filesystem system program_options)
+find_package(Boost 1.68 REQUIRED COMPONENTS serialization filesystem program_options)
 
 # on macOS we need to check whether to use libc++ or libstdc++ with clang++
 if(CMAKE_CXX_COMPILER_ID MATCHES "^(Apple)?Clang$")

--- a/CMakeModules/CPackSettings.cmake
+++ b/CMakeModules/CPackSettings.cmake
@@ -45,7 +45,7 @@ if(${CMAKE_SYSTEM_NAME} STREQUAL "Linux")
         OUTPUT_VARIABLE UBUNTU_RELEASE
         OUTPUT_STRIP_TRAILING_WHITESPACE)
     set(CPACK_PACKAGE_FILE_NAME "${PROJECT_NAME}_${PROJECT_VERSION}_${CPACK_DEBIAN_PACKAGE_ARCHITECTURE}-Ubuntu${UBUNTU_RELEASE}")
-    set(CPACK_DEBIAN_PACKAGE_DEPENDS "python${PYTHON_VERSION}, libboost-serialization-dev, libboost-filesystem-dev, libboost-system-dev, libboost-program-options-dev, libboost-test-dev, libtriangle-dev")
+    set(CPACK_DEBIAN_PACKAGE_DEPENDS "python${PYTHON_VERSION}, libboost-serialization-dev, libboost-filesystem-dev, libboost-program-options-dev, libboost-test-dev, libtriangle-dev")
 endif()
 
 if(WIN32)

--- a/CMakeModules/OMPLUtils.cmake
+++ b/CMakeModules/OMPLUtils.cmake
@@ -5,7 +5,6 @@ macro(add_ompl_test test_name)
     Boost::program_options
     Boost::serialization
     Boost::filesystem
-    Boost::system
     Boost::unit_test_framework)
   add_test(NAME ${test_name} COMMAND $<TARGET_FILE:${test_name}>)
 endmacro(add_ompl_test)

--- a/demos/CMakeLists.txt
+++ b/demos/CMakeLists.txt
@@ -12,7 +12,6 @@ if (OMPL_BUILD_DEMOS)
             ompl::ompl
             Eigen3::Eigen
             Boost::filesystem
-            Boost::system
             Boost::program_options)
     endmacro(add_ompl_demo)
 

--- a/install-ompl-ubuntu.sh.in
+++ b/install-ompl-ubuntu.sh.in
@@ -20,7 +20,6 @@ install_common_dependencies()
         libboost-filesystem-dev \
         libboost-program-options-dev \
         libboost-serialization-dev \
-        libboost-system-dev \
         libboost-test-dev \
         libeigen3-dev \
         libexpat1 \

--- a/omplConfig.cmake.in
+++ b/omplConfig.cmake.in
@@ -12,7 +12,7 @@ set_and_check(OMPL_INCLUDE_DIRS @PACKAGE_INCLUDE_INSTALL_DIR@)
 
 include ("${CMAKE_CURRENT_LIST_DIR}/omplExport.cmake" )
 include(CMakeFindDependencyMacro)
-set(_@PROJECT_NAME@_boost_components serialization filesystem system)
+set(_@PROJECT_NAME@_boost_components serialization filesystem)
 find_dependency(Boost REQUIRED COMPONENTS ${_@PROJECT_NAME@_boost_components})
 if(Boost_FOUND)
     foreach(_comp ${_@PROJECT_NAME@_boost_components})

--- a/scripts/docker/ompl.Dockerfile
+++ b/scripts/docker/ompl.Dockerfile
@@ -13,7 +13,6 @@ RUN apt-get update && \
         libboost-program-options-dev \
         libboost-python-dev \
         libboost-serialization-dev \
-        libboost-system-dev \
         libboost-test-dev \
         libeigen3-dev \
         libexpat1 \
@@ -58,7 +57,6 @@ RUN apt-get update && \
         libboost-program-options-dev \
         libboost-python-dev \
         libboost-serialization-dev \
-        libboost-system-dev \
         libeigen3-dev \
         libflann-dev \
         libtriangle-dev \

--- a/src/ompl/CMakeLists.txt
+++ b/src/ompl/CMakeLists.txt
@@ -44,7 +44,6 @@ target_link_libraries(ompl
     PUBLIC
         Boost::filesystem
         Boost::serialization
-        Boost::system
         Eigen3::Eigen
         "$<$<BOOL:${Threads_FOUND}>:Threads::Threads>"
         "$<$<BOOL:${OMPL_HAVE_FLANN}>:flann::flann>"

--- a/vcpkg.json
+++ b/vcpkg.json
@@ -7,7 +7,6 @@
         "boost-odeint",
         "boost-program-options",
         "boost-serialization",
-        "boost-system",
         "boost-test",
         "eigen3",
         "flann"

